### PR TITLE
BC-91 Addition tests for Node registration updates.

### PR DESCRIPTION
**Pull request by olegtropinin**

### CODE REVIEW

File: node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py [hunk 1]
```diff
@@ -54,35 +51,14 @@ def test_node_declaration_signed_change_request_as_primary_validator(api_client,
...
-    node = Node.objects.get_or_none(_id=regular_node.identifier)
+    assert blockchain_facade.get_next_block_number() == 2
+    node = blockchain_facade.get_node_by_identifier(regular_node.identifier)
     assert node
     assert node.identifier == regular_node.identifier
     assert node.fee == regular_node.fee
     assert node.addresses == regular_node.addresses
 
 
-@pytest.mark.django_db
```
Comment (dmugtasimov) [inline node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py:65]: @olegtropinin why removed?

File: node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py [hunk 2]
```diff
@@ -215,3 +191,51 @@ def test_node_declaration_scr_as_other_roles(
...
+        account_lock=primary_validator_node.identifier,
+    )
+    signed_change_request = SignedChangeRequest.create_from_signed_change_request_message(
+        message=message,
+        signing_key=primary_validator_key_pair.private,
+    )
+    payload = signed_change_request.json()
+    response = api_client.post('/api/signed-change-requests/', payload, content_type='application/json')
+    assert response.status_code == 201
+    assert response.json()['message']['node'] == updated_node
```
Comment (dmugtasimov) [inline node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py:241]: @olegtropinin this is not enough. You need to use blockchain facade methods to get node and assert for its values

File: node/core/tests/test_node_client.py [hunk 3]
```diff
@@ -38,16 +37,17 @@ def test_send_signed_change_request(
...
+    assert blockchain_facade.get_node_by_identifier(regular_node_declaration_signed_change_request.signer) is None
 
     client = smart_mocked_node_client
     scr = regular_node_declaration_signed_change_request
     response = client.send_signed_change_request(test_server_address, scr)
     assert response.status_code == 201
 
     assert BlockchainFacade.get_instance().get_next_block_number() == 2
-    assert ORMNode.objects.filter(_id=regular_node_declaration_signed_change_request.signer).exists()
+    assert blockchain_facade.get_node_by_identifier(regular_node_declaration_signed_change_request.signer) is not None
```
Comment (dmugtasimov) [inline node/core/tests/test_node_client.py:50]: @olegtropinin why changed? why can't we keep both asserts? the same questions for all other cases?

File: node/core/tests/test_node_client.py [hunk 4]
```diff
@@ -181,7 +181,7 @@ def test_yield_blocks_filtered(test_server_address, bloated_blockchain, smart_mo
 def test_get_node_online_address(test_server_address, smart_mocked_node_client, self_node):
     assert self_node.identifier == get_node_identifier()
     assert test_server_address in self_node.addresses
-    assert ORMNode.objects.filter(_id=self_node.identifier).exists()
```
Comment (dmugtasimov) [inline node/core/tests/test_node_client.py:184]: https://github.com/thenewboston-developers/Node/pull/175/files#r818522886

File: node/core/tests/test_node_client.py [hunk 5]
```diff
@@ -38,16 +37,17 @@ def test_send_signed_change_request(
 def test_send_scr_to_address_integration(
     test_server_address, regular_node_declaration_signed_change_request, smart_mocked_node_client
 ):
-    assert BlockchainFacade.get_instance().get_next_block_number() == 1
-    assert not ORMNode.objects.filter(_id=regular_node_declaration_signed_change_request.signer).exists()
+    blockchain_facade = BlockchainFacade.get_instance()
```
Comment (dmugtasimov) [inline node/core/tests/test_node_client.py:40]: https://github.com/thenewboston-developers/Node/pull/175/files#r818522886

File: node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py [hunk 6]
```diff
@@ -4,23 +4,20 @@
...
+from node.blockchain.types import NodeRole
 from node.core.utils.collections import deep_update
 
 
 @pytest.mark.usefixtures('base_blockchain', 'as_primary_validator')
 def test_node_declaration_signed_change_request_as_primary_validator(api_client, regular_node, regular_node_key_pair):
-    facade = BlockchainFacade.get_instance()
-    assert facade.get_next_block_number() == 1
-    assert not Node.objects.filter(_id=regular_node.identifier).exists()
+    blockchain_facade = BlockchainFacade.get_instance()
```
Comment (dmugtasimov) [inline node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py:18]: https://github.com/thenewboston-developers/Node/pull/175/files#r818522886

File: node/blockchain/tests/test_blockchain_facade/test_get_node_role.py [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
 from node.blockchain.models import Schedule
 from node.blockchain.types import NodeRole
 from node.core.utils.cryptography import get_node_identifier
 
 
 @pytest.mark.usefixtures('base_blockchain')
 def test_get_node_role_as_not_declared():
-    assert not ORMNode.objects.filter(_id=get_node_identifier()).exists()
-    assert BlockchainFacade.get_instance().get_node_role() is None
+    blockchain_facade = BlockchainFacade.get_instance()
```
Comment (dmugtasimov) [inline node/blockchain/tests/test_blockchain_facade/test_get_node_role.py:11]: https://github.com/thenewboston-developers/Node/pull/175/files#r818522886

File: node/core/tests/test_node_client.py (hunk 4 above)
Comment (olegtropinin) [reply node/core/tests/test_node_client.py:184]: I have added both checks.

File: node/core/tests/test_node_client.py (hunk 3 above)
Comment (olegtropinin) [reply node/core/tests/test_node_client.py:52]: I have kept both asserts.

File: node/core/tests/test_node_client.py (hunk 5 above)
Comment (olegtropinin) [reply node/core/tests/test_node_client.py:41]: I have kept both asserts.

File: node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py (hunk 2 above)
Comment (olegtropinin) [reply node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py]: Yes, I added blockchain facade method as you said.

**Review by olegtropinin — COMMENTED** [review]
@dmugtasimov 
I fixed all comments. Will you please check it?

File: node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py (hunk 1 above)
Comment (dmugtasimov) [reply node/blockchain/tests/test_rest_api/test_signed_change_request/test_node_declaration.py:65]: ok

**Pull request merged**